### ISSUE TITLE
Set operation filtering

### ISSUE THREAD

**Issue by dovej** [roost-im/roost-client#77]
This is the canonical example of something we more or less _need_ to wait on backend support for. Add support for set operation filtering. That is, class, instance, and recipient filtering are our atomic filters, now allow for unions, intersections, and negations of those atoms and so on.

After we have this, we can start considering our original plan to push a workflow that involves "popping out" when narrowing (hiding the narrowed thing from the mother pane), our implementation of punting and trashing, etc. A new issue for this workflow should be opened after this one is closed.
